coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,323 @@
+                              .select_related('status') \
+                              .prefetch_related('allocationattribute_set',
+                                                'resources'):
+                try:
+                    resource = allocation.resources.first().name
+                    style = (lambda x:
+                            self.style.WARNING('CURRENTLY EXPECTED BEHAVIOR: ') \
+                            + self.style.WARNING(x)) if \
+                            resource.endswith('Directory') \
+                            else self.style.ERROR
+                    if allocation.status.name != 'Expired':
+                        self.stdout.write(style(f'Project {project.name} is '
+                        f'inactive and has an unexpired '
+                        f'{resource} allocation {allocation.id}.'))
+                except AttributeError as e:
+                    pass
+                try:
+                    allocation_attribute = allocation.allocationattribute_set \
+                        .get(allocation_attribute_type__name='Service Units')
+                    if float(allocation_attribute.value) != 0:
+                        self.stdout.write(self.style.ERROR(
+                        f'Project {project.name} is inactive and has '
+                        f'non-zero SUs.'))
+                except AllocationAttribute.DoesNotExist as e:
+                    pass
+
+    def handle_project_pi(self):
+        '''
+        Assert that all projects have at least one PI.
+        '''
+        projects = Project.objects \
+            .select_related('status') \
+            .order_by('status__name', 'name')
+        for project in projects:
+            if not project.pis().exists():
+                self.stdout.write(self.style.ERROR( f'{project.status.name} '
+                f'Project {project.name} has no PIs.'))
+
+    def handle_user_project(self):
+        '''
+        Assert that all users with a cluster UID should be associated with
+        at least one Project.
+        '''
+        users = UserProfile.objects.filter(cluster_uid__isnull=False) \
+            .select_related('user') \
+            .order_by('user__is_active', 'user__username') \
+            .values('id', 'user__first_name', 'user__last_name',
+                    'user__email', 'user__username', 'user__is_active')
+
+        for user in users:
+            user_project_exists = ProjectUser.objects \
+                .filter(user_id=user['id']).exists()
+            if not user_project_exists:
+                self.stdout.write(self.style.ERROR(
+                f'{("Inactive", "Active")[user["user__is_active"]]} User '
+                f'{user["user__username"]} ({user["user__first_name"]} '
+                f'{user["user__last_name"]}, {user["user__email"]}) '
+                f'has a cluster UID but is not associated with any projects.'))
+
+    def handle_lrc_user_billing(self):
```

Comment:
We'll also want to check validity--see `coldfront.core.billing.utils.validation.is_billing_id_valid`.

This won't be testable on a dev. VM because the billing database is restricted.

(I'm also happy to write the LRC checks, since you don't have an LRC VM--let me know.)

Reply:
I think it'd be great if you wrote the LRC checks as I'm bound to deal with a bunch of errors trying it without a VM